### PR TITLE
feat: added dateTimeToLocaleString pipe

### PR DESCRIPTION
**Pull request by malua**
The PR adds the toLocaleString function from Luxon as an Angular pipe.

Please let me know if anythings missing or wrong :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,6 +210,25 @@ Transforms a DateTime into an arbitrarily formatted string:
...
+                                    year: 'numeric',
+                                    month: 'long',
+                                    day: 'numeric',
+                                    hour: 'numeric',
+                                    minute: '2-digit',
+                                    timeZoneName: 'short'
+                                  } }}
+```
+
+The pipe takes [`Intl.DateTimeFormat`][intl-datetimeformat] as the first parameter and Luxon´s [`LocaleOptions`][luxon-localeoptions] as an optional second parameter to override the configuration options on this DateTime.
```
Comment (dstelljes) [inline README.md:230]: None of the other pipes take `opts` right now, so let’s leave that off for now and add it to all pipes in a separate major change

File: projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { DateTime, LocaleOptions } from 'luxon';
+import { DateTimeToLocaleStringPipe } from './date-time-to-locale-string.pipe';
+
+describe('dateTimeToLocaleString', () => {
+  const format = DateTime.DATETIME_FULL;
+  const localeOptions: LocaleOptions = {locale: 'en-US'}
+  const pipe = new DateTimeToLocaleStringPipe();
+
+  describe('#transform', () => {
+    it('transforms an empty input into null', () => {
```
Comment (dstelljes) [inline projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts:10]: Add another test for null format

File: projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      expect(pipe.transform(null, format, localeOptions)).toBeNull();
+      expect(pipe.transform(undefined, format, localeOptions)).toBeNull();
+    });
+
+    it('transforms a DateTime into an arbitrarily formatted string', () => {
+      expect(pipe.transform(DateTime.fromISO('2006-01-02T15:04:05-07:00'), format, localeOptions))
+        .toBe("January 2, 2006 at 11:04 PM GMT+1");
```
Comment (dstelljes) [inline projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts:17]: This assertion only passes in GMT+1 🙂

**Review by dstelljes — CHANGES_REQUESTED** [review]

File: projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts (hunk 2 above)
Comment (malua) [reply projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts:9]: Thanks for your feedback. The thing is I don't know how to porperly assert the output, if I cannot specify the locale. Because depending on the system the output will be different.

e.g. if format is null, the output will either be MM/DD/YYYY or DD.MM.YYYY depending on the locale

Do you have an idea how to write this test?

File: README.md (hunk 1 above)
Comment (malua) [reply README.md]: Sure, updated the PR.

File: projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts (hunk 3 above)
Comment (malua) [reply projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts]: Here it's the same problem as above.

File: projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts (hunk 2 above)
Comment (dstelljes) [reply projects/luxon-angular/src/lib/formatting/date-time-to-locale-string.pipe.spec.ts:9]: In [other tests](https://github.com/dstelljes/luxon-angular/blob/main/projects/luxon-angular/src/lib/formatting/date-time-to-format.pipe.spec.ts#L16) we’ve used regular expressions to account for possible timezone differences, but yeah, that may not be possible here. Would using [`Date.toLocaleString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString) to get the expected output work?

**Review by dstelljes — APPROVED** [review]

**Comment by dstelljes** [comment]
Thanks for submitting! I’ll try to cut a release with this later today
